plugins/providers/alicloudiam/config.go

```diff
@@ -0,0 +1,228 @@
+package alicloudiam
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+
+	"github.com/bearaujus/bptr"
+	"github.com/go-playground/validator/v10"
+	"github.com/goto/guardian/domain"
+	"github.com/mitchellh/mapstructure"
+)
+
+const (
+	AccountTypeRamUser = "ramUser"
+	AccountTypeRamRole = "ramRole"
+)
+
+type Credentials struct {
+	AccessKeyID     string `mapstructure:"access_key_id" json:"access_key_id" validate:"required,base64"`
+	AccessKeySecret string `mapstructure:"access_key_secret" json:"access_key_secret" validate:"required,base64"`
+	RoleToAssume    string `mapstructure:"role_to_assume" json:"role_to_assume,omitempty"`
+	ResourceName    string `mapstructure:"resource_name" json:"resource_name" validate:"required"`
+}
+
+func (c *Credentials) Encrypt(encryptor domain.Encryptor) error {
+	if c == nil {
+		return ErrUnableToEncryptNilCredentials
+	}
+
+	encryptedAccessKeyID, err := encryptor.Encrypt(c.AccessKeyID)
+	if err != nil {
+		return err
+	}
+
+	encryptedAccessKeySecret, err := encryptor.Encrypt(c.AccessKeySecret)
+	if err != nil {
+		return err
+	}
+
+	c.AccessKeyID = encryptedAccessKeyID
+	c.AccessKeySecret = encryptedAccessKeySecret
+	return nil
+}
+
+func (c *Credentials) Decrypt(decryptor domain.Decryptor) error {
+	if c == nil {
+		return ErrUnableToDecryptNilCredentials
+	}
+
+	decryptedAccessKeyID, err := decryptor.Decrypt(c.AccessKeyID)
+	if err != nil {
+		return err
+	}
+
+	decryptedAccessKeySecret, err := decryptor.Decrypt(c.AccessKeySecret)
+	if err != nil {
+		return err
+	}
+
+	c.AccessKeyID = decryptedAccessKeyID
+	c.AccessKeySecret = decryptedAccessKeySecret
+	return nil
+}
+
+type Permission struct {
```

Comment:
since the list of permissions are going to be stored in the `appeals`, and `grants` db, this has to implement `.String()` method as well, you can take reference from our existing tableau plugin here https://github.com/goto/guardian/blob/df093f61c8caf387f749046eea6d6c91e8d5323d/plugins/providers/tableau/config.go#L66-L88

Reply:
Resolved: https://github.com/goto/guardian/pull/190/commits/6bd8fdffe0d3723f936bc84ddaf326d111b06d9b

Reply:
Hi pak @rahmatrhd .. After validating the changes using this method. I think String() is not triggered

// This image is example policy on my local
![image](https://github.com/user-attachments/assets/4bce3501-8878-433c-8754-32baeaa91de0)

But when I using my old method it was working fine.

Reply:
[TESTED] the `String() string` is not triggered. So I will remove this function 🙏